### ISSUE TITLE
🐛 Pass auth config to node requests for fonts

### ISSUE THREAD

**Issue by Robdel12** [percy/cli#969]
## What is this?

In #921 we realized font responses were being corrupted through the browsers request interception API. To get around that, we now request and save fonts straight from node as a binary (no encoding). In that PR, I forgot to pass the auth config to the request that's being made, which will prevent sites behind auth from capturing fonts properly. 

To fix this we need to pass the config we already have with the node request so it's properly authenticated. We should also add a debug log for when we're requesting fonts via node.

**Comment by mir3z** [comment]
I guess that's the reason why we are getting

```
11:03:34  [percy:core:discovery] Error: self signed certificate
11:03:34      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
11:03:34      at TLSSocket.emit (node:events:520:28)
11:03:34      at TLSSocket._finishInit (node:_tls_wrap:944:8)
11:03:34      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (1ms)
```

while fetching fonts. Any ETA for the fix @Robdel12 ?

**Comment by Robdel12** [comment]
👋🏼 hey @mir3z! I don't think that's related. If you could share the logs around that, I think it'd tell more of a story. `Error: self signed certificate` typically means there's an invalid certificate in the chain and fails to connect to our API

**Comment by mir3z** [comment]
> @Robdel12 Thanks for the response. Indeed we use self-signed ceritficate but that wasn't a problem until we updated `@percy/cypress@2.3.2` to `@percy/cypress@3.0.0` + `@percy/cli@1.4.0`.
> 
> What's strange for me is that this error occurs only for fonts. Other static assets are discovered and fetched correctly.
> 
> See more logs, I've redacted some labels with `<###>`:
> 
> ```
> 21:52:02  [percy:core:snapshot] --------- (1649ms)
> 21:52:02  [percy:core:snapshot] Handling snapshot: <###> (0ms)
> 21:52:02  [percy:core:snapshot] - url: https://localhost:3003/#/<###> (0ms)
> 21:52:02  [percy:core:snapshot] - widths: 1280px (0ms)
> 21:52:02  [percy:core:snapshot] - minHeight: 1024px (0ms)
> 21:52:02  [percy:core:snapshot] - discovery.allowedHostnames: localhost (1ms)
> 21:52:02  [percy:core:snapshot] - discovery.disableCache: true (0ms)
> 21:52:02  [percy:core:snapshot] - clientInfo: @percy/cypress/3.0.0 (0ms)
> 21:52:02  [percy:core:snapshot] - environmentInfo: cypress/5.1.0 (0ms)
> 21:52:02  [percy:core:snapshot] - domSnapshot: true (0ms)
> 21:52:02  [percy:core:page] Page created (13ms)
> 21:52:02  [percy:core:page] Resize page to 1280x1024 (120ms)
> 21:52:02  [percy:core:page] Navigate to: https://localhost:3003/#/<###>(11ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/ (5ms)
> 21:52:02  [percy:core:discovery] - Serving root resource (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/css/typography.css (45ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/css/reset.css (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/css/bundle.css (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/SourceSansPro-Regular.woff2 (1ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/SourceSansPro-Regular.woff (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/SourceSansPro-SemiBold.woff2 (1ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/SourceSansPro-SemiBold.woff (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/SourceSansPro-Bold.woff2 (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/SourceSansPro-Bold.woff (1ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/Material-Design-Iconic-Font.woff2 (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/fonts/Material-Design-Iconic-Font.woff (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/css/mixpanel.css (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/images/<###>.svg (0ms)
> 21:52:02  [percy:core:discovery] Handling request: https://localhost:3003/images/<###>/<###>.png (1ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/css/reset.css (55ms)
> 21:52:02  [percy:core:discovery] - sha: b3416ca29461a58863464125ebdfc36a916b8df3604853f0246e287acb22a54a (0ms)
> 21:52:02  [percy:core:discovery] - mimetype: text/css (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/css/mixpanel.css (0ms)
> 21:52:02  [percy:core:discovery] - sha: 86b19a7bb7e20543c88e94705805b131b4ac0f9cf81b3eb69b4e8a2b7479c345 (0ms)
> 21:52:02  [percy:core:discovery] - mimetype: text/css (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/css/typography.css (44ms)
> 21:52:02  [percy:core:discovery] - sha: 65cf3ede17f6e57fc10e4a5bc4467e5ab4d14b94c8009227ad8445050d19e9d9 (0ms)
> 21:52:02  [percy:core:discovery] - mimetype: text/css (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/SourceSansPro-Regular.woff2 (2ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/SourceSansPro-Regular.woff2 (11ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/SourceSansPro-Regular.woff (52ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/SourceSansPro-Regular.woff (9ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/css/bundle.css (12ms)
> 21:52:02  [percy:core:discovery] - sha: c79c24706f5c693cc1d5bc4fd7177278310af4a68afd0b564af649157695d2aa (1ms)
> 21:52:02  [percy:core:discovery] - mimetype: text/css (1ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/Material-Design-Iconic-Font.woff2 (88ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/Material-Design-Iconic-Font.woff2 (7ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/Material-Design-Iconic-Font.woff (3ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/images/<###>.svg (1ms)
> 21:52:02  [percy:core:discovery] - sha: 80e534817f7b1d73e3a59d1163d9bc9015d01cdac4f776c35edeef001b63394f (0ms)
> 21:52:02  [percy:core:discovery] - mimetype: image/svg+xml (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/SourceSansPro-SemiBold.woff2 (3ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/Material-Design-Iconic-Font.woff (9ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/SourceSansPro-SemiBold.woff2 (8ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/SourceSansPro-SemiBold.woff (2ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/SourceSansPro-Bold.woff2 (11ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/SourceSansPro-SemiBold.woff (3ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/fonts/SourceSansPro-Bold.woff (3ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/SourceSansPro-Bold.woff2 (8ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:page] Page navigated (0ms)
> 21:52:02  [percy:core:network] Wait for 250ms idle (0ms)
> 21:52:02  [percy:core:discovery] Encountered an error processing resource: https://localhost:3003/fonts/SourceSansPro-Bold.woff (12ms)
> 21:52:02  [percy:core:discovery] Error: self signed certificate
> 21:52:02      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
> 21:52:02      at TLSSocket.emit (node:events:520:28)
> 21:52:02      at TLSSocket._finishInit (node:_tls_wrap:944:8)
> 21:52:02      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) (0ms)
> 21:52:02  [percy:core:discovery] Processing resource: https://localhost:3003/images/<###>/<###>.png (36ms)
> 21:52:02  [percy:core:discovery] - sha: 238872f55d64ec2df6817726c262834e3cacede30819af26532c34fd109d59bc (1ms)
> 21:52:02  [percy:core:discovery] - mimetype: image/png (0ms)
> 21:52:03  [percy:core] Snapshot taken: <###> (253ms)
> 21:52:03  [percy:core:page] Page closed (9ms)
> ```

@Robdel12 Did you have a chance to take a look at the logs? Is there any way I could help investigating this issue further?

**Comment by Robdel12** [comment]
@kwaves This should be fixed in v1.6.1 now https://github.com/percy/cli/releases/tag/v1.6.1 

Thanks for reporting that!
